src/components/FileTreeNode/index.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { TreefoldRenderProps } from 'react-treefold';
+import { ListGroup } from 'react-bootstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import makeClassName from 'classnames';
+
+import { gettext } from '../../utils';
+import { TreeNode } from '../FileTree';
+import styles from './styles.module.scss';
+
+type PublicProps = TreefoldRenderProps<TreeNode>;
+
+class FileTreeNodeBase<TreeNodeType> extends React.Component<PublicProps> {
+  render() {
+    const {
+      node,
+      level,
+      isFolder,
+      isExpanded,
+      getToggleProps,
+      hasChildNodes,
+      renderChildNodes,
+    } = this.props;
+
+    let listGroupItemProps = {
+      className: makeClassName(styles.node, {
+        [styles.directoryNode]: isFolder,
+      }),
+    };
+
+    if (isFolder) {
+      listGroupItemProps = {
+        ...listGroupItemProps,
+        ...getToggleProps(),
+      };
+    }
+
+    return (
+      <React.Fragment>
+        <ListGroup.Item {...listGroupItemProps}>
+          <span style={{ paddingLeft: `${level * 20}px` }}>
+            {isFolder ? (
+              <React.Fragment>
+                <FontAwesomeIcon icon={isExpanded ? 'folder-open' : 'folder'} />
```

Comment:
I'm not sure I love `react-fontawesome`. What does it buy us over using `Font Awesome` with just basic classes and css? This seems like a lot of code to be adding just to be able to use a React component instead of assigning a className. If we'd need a `span` or something else in its place then I guess it makes sense, but would it be possible to do something just with css using `::before`?

Reply:
FA allows us to use lots of icons easily. We don't have UX so I don't want to pull random icons and end up with an inconsistent UI. Bootstrap and FA work well together.

We will likely need some other icons for some metadata, maybe for the errors vs warnings, links, etc. FA has all of that, the license is permissive and, again, all icons are consistent.

Also, those icons work on all devices/OS/browsers 😛

Reply:
My issue wasn’t with the use of font awesome, but with the react package and using a component to render the icon. I thought maybe we could use font awesome in conjunction with css to add icons to existing elements rather than needing to include an extra component.

Reply:
ah, ooops. It's true that the setup is weird but it's the recommended react way.. We could maybe create our own component instead but I am not sure if that'd be better.
I haven't checked but it is possible that the lib inlines the SVGs.